### PR TITLE
Some attributes to ios_application are labels that cannot be empty str

### PR DESCRIPTION
**Pull request by amberdixon**
https://github.com/bazelbuild/rules_apple/blob/master/doc/rules-ios.md#ios_application lists out the attributes to ios_applicaiton and whether or not they must be Labels.

If they are labels, they cannot be empty strings. If an empty string is supplied to a label argument, convert it to nil (which will convert to Starlark's None)

### CODE REVIEW

File: spec/integration/monorepo/after/.bazelversion [hunk 1]
```diff
@@ -0,0 +1 @@
+3.7.0
```
Comment (thiagohmcruz) [inline spec/integration/monorepo/after/.bazelversion:1]: Can you just clarify why this change?

**Review by thiagohmcruz — APPROVED** [review]
LGTM!

**Review by congt — APPROVED** [review]

File: lib/cocoapods/bazel/target.rb [hunk 2]
```diff
@@ -172,8 +172,8 @@ def resolved_value_by_build_setting(setting, additional_settings: {})
           debug_value
```
Comment (segiddins) [inline lib/cocoapods/bazel/target.rb:172]: I think we also want to apply the logic in the non-select case?

File: spec/integration/monorepo/after/.bazelversion (hunk 1 above)
Comment (amberdixon) [reply spec/integration/monorepo/after/.bazelversion:1]: Rules_ios does not work on bazel 4.0.0, because it relies on a method that is deprecated in the newer version.

**Review by segiddins — APPROVED** [review]
